packages/api/test/e2e/staking.e2e.ts

```diff
@@ -0,0 +1,304 @@
+    const minimumBond = 1_234;
+    const setMinimumBondTx = api.tx.staking.setMinimumBond(minimumBond);
+
+    await api.tx.sudo.sudo(setMinimumBondTx).signAndSend(alice, async ({ status }) => {
+      if (status.isInBlock) {
+        expect((await api.query.staking.minimumBond()).toString()).toEqual(minimumBond.toString());
+        done();
+      }
+    });
+  });
+
+
+  test('Set invulnerable validators', async done => {
+    const invulnerables: AccountId[] = keyring.getPairs().map(p => p.publicKey as AccountId);
+    const setInvulnerablesTx = api.tx.staking.setInvulnerables(invulnerables);
+
+    await api.tx.sudo.sudo(setInvulnerablesTx).signAndSend(alice, async ({ status }) => {
+      if (status.isInBlock) {
+        expect(
+          (await api.query.staking.invulnerables()).map(k => keyring.encodeAddress(k))
+        ).toEqual(
+          invulnerables.map(k => keyring.encodeAddress(k))
+        );
+        done();
+      }
+    });
+  });
+
+  test('Force no eras', async done => {
+    await api.query.staking.forceEra(
+      (forcing: Forcing) => {
+        if (forcing.isForceNone) done();
+      }
+    );
+    await api.tx.sudo.sudo(api.tx.staking.forceNoEras()).signAndSend(alice);
+  });
+
+  test('Force new era', async done => {
+    await api.query.staking.forceEra(
+      (forcing: Forcing) => {
+        if (forcing.isForceNew) done();
+      }
+    );
+    await api.tx.sudo.sudo(api.tx.staking.forceNewEra()).signAndSend(alice);
+  });
+
+  test('Force new era always', async done => {
+    await api.query.staking.forceEra(
+      (forcing: Forcing) => {
+        if (forcing.isForceAlways) done();
+      }
+    );
+    await api.tx.sudo.sudo(api.tx.staking.forceNewEraAlways()).signAndSend(alice);
+  });
+
+  test('Force unstake', async done => {
+    // Use charlie account as bob stash, it's simpler than funding a new account.
+    const bob_stash = keyring.addFromUri('//Charlie');
+    // bond bob's stash account.
+    await api.tx.staking.bond(bob.address, 10_000, 'controller').signAndSend(
+      bob_stash,
+      async ({ status }) => {
+        if (status.isInBlock) {
+          const controller = (await api.query.staking.bonded(bob_stash.address)) as Option<AccountId>;
+          expect(controller.unwrapOr(null)).toBeDefined();
+          done();
+        }
+      });
+
+    const unstake = api.tx.staking.forceUnstake(bob_stash.address);
+    await api.tx.sudo.sudo(unstake).signAndSend(alice, async ({ status }) => {
+      if (status.isInBlock) {
+        // bob stash is removed / unbonded
+        const controller = (await api.query.staking.bonded(bob_stash.address)) as Option<AccountId>;
+        expect(controller.unwrapOr(null)).toBeNull();
+        done();
+      }
+    });
+  });
+
+  /// Cancel enactment of a deferred slash. Can be called by root origin
+  /// passing the era and indices of the slashes for that era to kill.
+  it.skip('Cancel deferred slash', async () => { });
+
```

Comment:
planning to look into the slashing stuff separately so I'll remove this from the PR